### PR TITLE
Onboarding validation

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This updates the member validation to include validating the onboarding fields, which potentially returns multiple errors.

Fixes SC-20543

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

- All non-empty fields should be validated, if the field is empty then it's not validated.
- Should only return True/No errors if all fields are provided and valid (there are no optional fields)
- Multiple errors may be returned, the backend won't know anything about onboarding order or what step the user is on

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [x] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/db/errors.go [hunk 1]
```diff
@@ -49,22 +54,44 @@ var (
 	ErrTopicNameTooLong          = errors.New("topic name is too long")
 )
 
+// ValidationError represents a validation error for a specific field and may contain
+// an index if the field is a slice.
 type ValidationError struct {
-	err error
+	Field string
+	Err   error
+	Index int
```
Comment (bbengfort) [inline pkg/tenant/db/errors.go:62]: This data structure will have to be JSON serialized for the front end:

```suggestion
	Field string `json:"field"`
	Err   error `json:"error"
	Index int `json:"index"
```

File: pkg/tenant/db/errors.go [hunk 2]
```diff
@@ -49,22 +54,44 @@ var (
...
+}
+
+func (e *ValidationError) AtIndex(index int) *ValidationError {
+	e.Index = index
+	return e
 }
 
 func (e *ValidationError) Error() string {
-	return fmt.Sprintf("validation error: %s", e.err)
+	return fmt.Sprintf("invalid field '%s': %s", e.Field, e.Err)
```
Comment (bbengfort) [inline pkg/tenant/db/errors.go:78]: Should we handle the case where there is no field or the case where there is an index in the string?

File: pkg/tenant/db/errors.go [hunk 3]
```diff
@@ -49,22 +54,44 @@ var (
...
+}
+
+type ValidationErrors []*ValidationError
+
+func (v ValidationErrors) Error() string {
+	errs := make([]string, 0, len(v))
+	for _, e := range v {
+		errs = append(errs, e.Error())
+	}
+	return fmt.Sprintf("%d validation errors occurred:\n%s", len(v), strings.Join(errs, "\n"))
```
Comment (bbengfort) [inline pkg/tenant/db/errors.go:96]:
```suggestion
	return fmt.Sprintf("%d validation errors occurred:\n  -%s", len(v), strings.Join(errs, "\n  -"))
```

A bit nicer formatting suggestion

**Review by bbengfort — APPROVED** [review]
Nice work! I like the new `ValidationError` data structure. My only concern is that the ValidationError needs to be serialized as JSON in the response; does `api.ErrorResponse` handle this type correctly?

File: pkg/tenant/db/errors.go (hunk 1 above)
Comment (pdeziel) [reply pkg/tenant/db/errors.go:62]: I was thinking of adding a similar struct to the API package, except that it contains the JSON tags for serialization, and then converting the one here into the API response struct. But if that's too clunky we could just add the tags here.

Comment (bbengfort) [reply pkg/tenant/db/errors.go:62]: That makes sense to me - whatever you'd prefer; just making sure I wasn't missing something in translation!

**Pull request merged**